pkg/resources/custom_diffs.go

```diff
@@ -270,3 +270,15 @@ func RecreateWhenStreamIsStale() schema.CustomizeDiffFunc {
 		return nil
 	}
 }
+
+// TODO: [SNOW-1763442] unable to test now, as there is no test accounts with different regions
+// RecreateWhenSecondaryConnectionChangedExternally detects if the secondary connection was promoted externally to serve as primary.
+// If so, it sets the `is_primary` field to `false` which is our desired value for secondary_connection
+func RecreateWhenSecondaryConnectionPromotedExternally() schema.CustomizeDiffFunc {
```

Comment:
nit: we have to handle this the other way around too for primary connection demotion